packages/editor/src/hooks/anchor.js

```diff
@@ -98,7 +98,7 @@ export const withInspectorControl = createHigherOrderComponent( ( BlockEdit ) =>
  */
 export function addSaveProps( extraProps, blockType, attributes ) {
 	if ( hasBlockSupport( blockType, 'anchor' ) ) {
-		extraProps.id = attributes.anchor;
+		extraProps.id = attributes.anchor === '' ? null : attributes.anchor;
```

Comment:
Alternatively, could this have been considered as part of the condition for assigning `id`?

i.e.

```js
if ( hasBlockSupport( blockType, 'anchor' ) && attributes.anchor ) {
```